docs/arcadia-users-group/20240416-intro-to-ml1/make_peptides_ml_dataset.md

```diff
@@ -0,0 +1,278 @@
+   peptide_bioactivities_concatenated <- peptide_bioactivity_with_clusters_long %>%
+     filter(value == 1) %>%
+     filter(rep == member) %>%
+     select(peptipedia_peptide_id = rep, bioactivity, value) %>%
+     group_by(peptipedia_peptide_id) %>%
+     summarise(all_bioactivities = str_c(bioactivity, collapse = ";"), .groups = 'drop')
+
+
+   # Join this back to your original (or relevant) dataset to add the concatenated bioactivities column
+   final_dataset <- peptide_group_bioactivities_filtered %>%
+     left_join(peptide_bioactivities_concatenated, by = "peptipedia_peptide_id") %>%
+     distinct()
+   
+   # join with other peptide metadata ----------------------------------------
+   
+   gene_ontology <- read_csv("~/Downloads/backup_peptipedia_29_03_2023/table_public.gene_ontology.csv",
+                             col_names = c("peptideid", "go_id", "go_description", "go_category")) %>%
+     select(peptideid, go_id)
+
+   pfam <- read_csv("~/Downloads/backup_peptipedia_29_03_2023/table_public.pfam.csv",
+                    col_names = c("peptideid", "pfam_id", "pfam_name", "pfam_type"))
+
+   final_dataset <- final_dataset %>%
+     left_join(gene_ontology, by = c("peptipedia_peptide_id" = "peptideid")) %>%
+     left_join(pfam, by = c("peptipedia_peptide_id" = "peptideid")) %>%
+     distinct()
+
+   final_dataset %>% group_by(group_bioactivity) %>% tally() %>% arrange(desc(n)) %>% knitr::kable()
+
+   write_tsv(final_dataset, "~/Downloads/backup_peptipedia_29_03_2023/peptides_for_ml_tmp2.tsv")
+   ```
+
+
+6. Convert the output TSV to a FASTA file that can be used to calculate peptide characteristics (molecular weight, etc).
+   ```bash
+   cut -f1,2 peptides_for_ml_tmp.tsv | tail -n +2 | seqkit tab2fx > peptides_for_ml.fasta
+   ```
+
+7. Calculate peptide characteristics (length, mw, etc.)
+
+   ```bash
+   curl https://raw.githubusercontent.com/Arcadia-Science/peptigate/main/scripts/characterize_peptides.py
+   python characterize_peptides.py peptides_for_ml.fasta peptides_for_ml_characterized.tsv
+   ```
+
+8. Combine peptide metadata with other information
+
+   ```r
+   peptide_characterization <- read_tsv("~/Downloads/backup_peptipedia_29_03_2023/peptides_for_ml_characterized.tsv")
+   
+   final_dataset <- final_dataset %>%
+     left_join(peptide_characterization, by = c("peptipedia_peptide_id" = "peptide_id")) %>%
+     mutate(length = nchar(peptide_sequence)) %>%
+     rename(bioactivity = group_bioactivity)
+ 
+   write_tsv(final_dataset, "peptides_ml_dataset.tsv")
+   ```
+
+9. Check that the data set is predictive of the label "group_bioactivity".
+   ```r
+   library(caret)
+   
+   set.seed(123)
+   
+   final_dataset <- read_tsv("peptides_ml_dataset.tsv")
+   final_dataset_filtered <- final_dataset %>%
+     select(-peptipedia_peptide_id, -peptide_sequence, -all_bioactivities, -go_id, -pfam_id,
+           -pfam_name, -pfam_type)
```

Comment:
why are these removed? should the pfam data be loaded in the first place?

Reply:
PFAM and GO are removed because there are a lot of NAs, which random forests (caret) in R can't handle. I added them to the data/left them in there so we can teach about missing data and have more character data in the data set.

At the bottom of the MD file, I made some notes on how we could extend the data set to have more character data where we wouldn't have missing data -- BLAST, taxonomy, other annotation, etc. I haven't done that here bc of time/wanting to get a first product into everyone's hands so that we could start lesson dev, but we/I could still do it.